ariadne/cardano/src/Ariadne/MainTemplate.hs

```diff
@@ -169,23 +169,21 @@ initializeEverything MainSettings {..}
     initAction :: IO ()
     initAction = walletInitAction
 
-    postStartupAction :: IO ()
-    postStartupAction = postInitAction
-
     serviceAction :: IO ()
     serviceAction =
       raceWithUpdateCheckAction cardanoAction
 
     mainAction :: IO ()
     mainAction = do
       initAction
+      withAsync postInitAction $ \postInitActionThread -> do
+        linkUI postInitActionThread $ msPutBackendErrorToUI uiFace
       -- Spawn backend actions in async thread, then run ui action in the main thread
       -- This is needed because some UI libraries (Qt) insist on livng in the main thread
-      withAsync (concurrently postStartupAction serviceAction) $ \serviceThread -> do
+        withAsync serviceAction $ \serviceThread -> do
           -- Make custom link to service thread for UI apps.
           -- It is going to call msPutBackendErrorToUI and rethrow exception, if something goes wrong
           linkUI serviceThread $ msPutBackendErrorToUI uiFace
-          logDebug logging "Launching the UI..."
```

Comment:
Why? To be honest, I don't understand why there are any changes in this file. Yes, some changes were necessary in a previous version, but now I think this file should remain intact.